### ISSUE TITLE
Multiple Rmd files in subdirectories and common lib_dir

### ISSUE THREAD

**Issue by gaborcsardi** [rstudio/rmarkdown#146]
I have a bunch of Rmd files, spread in some subdirectories, e.g. 

```
aaa/bbb/one.Rmd
aaa/ccc/two.Rmd
...
```

I want to convert them to HTML, and use a common `lib_dir` for all of them. The problem is, that if I do

```
render("aaa/bbb/one.Rmd", NULL, 
       output_options=list(self_contained=FALSE, lib_dir="lib"))
```

then the `lib` directory (and more importantly the the references in the generated HTML accordingly) end up in `aaa/bbb/lib`, which is obviously not what I want.

If I try

```
setwd("aaa/bbb/")
render("one.Rmd", NULL, output_options=list(self_contained=FALSE, lib_dir="../../lib"))
```

then I get

```
Error in relativeTo(basepath, dir) : 
  The path /private/tmp/lib/jquery-1.11.0 does not appear to be a descendant of /private/tmp/aaa/bbb/
```

Is there a way to create a common `lib_dir` for these files, and still get the links in the HTML files right?

**Comment by sanjmeh** [comment]
### This is a strange restriction. 
If I have understood it correct, the `lib_dir` is expected to be in the same directory as `output_dir` argument of `rmarkdown::render()`. This is a big limitation as it forces all the output html files to be saved in the same parent directory where the `.Rmd` is present. If we want to use the same `.Rmd` to generate a separate bunch of html files (based on the argument `output_dir`), we cannot share the same `lib_dir`. I have 50+ html files generated in every pass, and cannot abandon the subdirectory structure. Do we have a solution to this?

**Comment by jonathan-g** [comment]
I have the same issue.

I organize my reveal.js slides for a semester in a directory structure:

    semester:
        lecture_lib (common material: css, js, etc.)
            assets (custom css themes, custom js, etc.)
            revealjs
            font-awesome
        slides
            class_01
                assets (per-class assets)
                    images (images to include)
                    videos (video files)
                    fig (figures generated by RMarkdown)
            class_02
                assets (per-class assets)
                    images (images to include)
                    videos (video files)
                    fig (figures generated by RMarkdown)
            c;lass_03
                assets (per-class assets)
                    images (images to include)
                    videos (video files)
                    fig (figures generated by RMarkdown)
            ...

This lets me keep a consistent structure for each lecture.

This worked fine with RMarkdown for several years until Pandoc 2.9, when modifications to html_dependencies() broke my structure because RMarkdown insists on creating a file header-attrs.js and I am not given an option of specifying a path to a static version of header-attrs.js.

If I use pandoc 2.7.x, RMarkdown works fine with my directory structures, but if I use any later version of Pandoc I get the same error message about relativeTo().

**Comment by cderv** [comment]
@jonathan-g , this is a pretty old issue (2014). 
Can you open a new one (linking to this one if you think this is related) with a reproducible example (would be ideal) ? 

The fact that something broke with new pandoc make me thing there is something to fix. 

Thank you !

**Comment by jonathan-g** [comment]
Sure.

**Comment by sjackman** [comment]
Could you please post a link to the new issue here?

**Comment by jonathan-g** [comment]
I just submitted a new issue #1859. I took some time to investigate this more carefully and to write up the new issue carefully before submitting it. I think a small PR can enhance rmarkdown to support this kind of non-standard directory structure (see the fork at https://github.com/jonathan-g/rmarkdown/tree/minimal-tree-fix).

**Comment by yihui** [comment]
This has finally been addressed in #2632. `rmarkdown` no longer errors when `lib_dir` points outside the output directory (e.g. `lib_dir = "../lib"`), so multiple `.Rmd` files in sibling subdirectories can share a common library directory — exactly the layout you described. The dependencies are now referenced with an up-tree relative path (`../lib/...`) instead of failing with "does not appear to be a descendant of". No new option is needed; it just works. Thanks for the original report.
